modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/SmileDeltaTermStructureParameters.java

```diff
@@ -5,479 +5,202 @@
+  default public SmileAndBucketedSensitivities getSmileAndSensitivitiesForTime(
+      double expiry,
+      DoubleArray volatilityAtTimeSensitivity) {
+
+    int nbVol = getNumberStrike();
+    ArgChecker.isTrue(volatilityAtTimeSensitivity.size() == nbVol, "Sensitivity with incorrect size");
     ArgChecker.isTrue(nbVol > 1, "Need more than one volatility value to perform interpolation");
-    int nbTime = _timeToExpiry.length;
+    int nbTime = getNumberSmiles();
     ArgChecker.isTrue(nbTime > 1, "Need more than one time value to perform interpolation");
     double[] volatilityT = new double[nbVol];
     double[][] volatilitySensitivity = new double[nbTime][nbVol];
     for (int loopvol = 0; loopvol < nbVol; loopvol++) {
       double[] volDelta = new double[nbTime];
       for (int looptime = 0; looptime < nbTime; looptime++) {
-        volDelta[looptime] = _volatilityTerm[looptime].getVolatility().get(loopvol);
+        volDelta[looptime] = getVolatilityTerm().get(looptime).getVolatility().get(loopvol);
       }
-      Interpolator1DDataBundle interpData =
-          _timeInterpolator.getDataBundleFromSortedArrays(_timeToExpiry, volDelta);
-      double[] volatilitySensitivityVol = _timeInterpolator.getNodeSensitivitiesForValue(interpData, time);
+      BoundCurveInterpolator bound = getTimeInterpolator().bind(
+          getTimeToExpiry(), DoubleArray.copyOf(volDelta), getTimeLeftExtrapolator(), getTimeRightExtrapolator());
+      double[] volatilitySensitivityVol = bound.parameterSensitivity(expiry).toArray();
       for (int looptime = 0; looptime < nbTime; looptime++) {
-        volatilitySensitivity[looptime][loopvol] = volatilitySensitivityVol[looptime] * volatilityAtTimeSensitivity[loopvol];
+        volatilitySensitivity[looptime][loopvol] =
+            volatilitySensitivityVol[looptime] * volatilityAtTimeSensitivity.get(loopvol);
       }
-      volatilityT[loopvol] = _timeInterpolator.interpolate(interpData, time);
+      volatilityT[loopvol] = bound.interpolate(expiry);
     }
-    SmileDeltaParameters smile = SmileDeltaParameters.of(time, _volatilityTerm[0].getDelta(), DoubleArray.copyOf(volatilityT));
+    SmileDeltaParameters smile = SmileDeltaParameters.of(expiry, getDelta(), DoubleArray.copyOf(volatilityT));
     return SmileAndBucketedSensitivities.of(smile, DoubleMatrix.copyOf(volatilitySensitivity));
   }
 
   /**
-   * Gets the name of smile parameter term structure.
+   * Gets the number of smiles.
    * 
-   * @return the name
+   * @return the number of smiles
    */
-  public String getName() {
-    return _name;
+  default int getNumberSmiles() {
+    return getVolatilityTerm().size();
   }
 
   /**
-   * Gets the times to expiry.
-   * @return The times.
+   * Gets the number of strikes. 
+   * 
+   * @return the number of strikes
    */
-  public double[] getTimeToExpiry() {
-    return _timeToExpiry;
+  default int getNumberStrike() {
```

Comment:
`Curve` uses `getParameterCount()`. Maybe this should be `getStrikeCount()` and `getSmileCount()`